vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
 description: Concepts about vcpkg registries and their capabilities.
 author: vicroms
 ms.author: viromer
-ms.date: 01/10/2024
+ms.date: 11/10/2024
 ms.topic: concept-article
 ---
-
 # Registries concepts
 
-Registries are collections of ports and their versions. The curated registry is
-the one hosted at <https://github.com/Microsoft/vcpkg>. vcpkg lets you create
-custom registries, which may be hosted by a variety of public or private providers.
+## Overview
+
+vcpkg hosts a selection of libraries packaged into [ports](ports.md) at
+<https://github.com/Microsoft/vcpkg>, this collection of ports is called the
+*curated registry*. However, vcpkg is not limited to just installing ports from
+the curated registry. Users are capable of extending the selection of ports by
+creating custom registries.
+
+A registry is a collection of ports and helper files arranged in a specific
+structure. By following the registry structure, ports contained in a registry
+can be installed with the same amount of features
+([versioning](../users/versioning.md), [binary
+caching](../users/binarycaching.md)) offered to ports in the curated registry.
+
+There are currently three kinds of registries: 
+
+* The [built-in registry](#built-in-registry),
+* [Git registries](#git-registries), and 
+* [filesystem registries](#filesystem-registries).
+
+vcpkg can be instructed to consider ports available in custom registries by
+using a [`vcpkg-configuration.json`
+file](../reference/vcpkg-configuration-json.md#registries). See the [Tutorial:
+Install a dependency from a Git-based registry](../consume/git-registries.md)
+article for a tutorial on how to use custom registries in your projects.
+
+## Registry structure
+
+For vcpkg to interface with a registry, it must comform with the structure
+below:
+
+* A directory named `ports`, contains the collection of ports, with each
+  subdirectory containing a specific port matching the subdirectory name. For
+  example, the files for port `foo` must be located in `ports/foo`.
+* A directory named `versions` must contain the files that comprise the [versions
+  database](#versions-database).
+  
+### Example: registry structure
+
+```
+ports/
+  foo/
+    portfile.cmake
+    vcpkg.json
+versions/
+  f-/
+    foo.json
+  baseline.json
+```
+
+## Versions database
+
+All registries, regardless of their kind, must contain a `versions` folder at the
```

Comment:
Given that we are describing to users what registries are, rather than describing what behavior something implementing registries must do, I think we should describe them with present tense, as in "registries contain a versions directory" rather than "registries must contain a versions directory". I think we should reserve 'must' for things the user has to do rather than things vcpkg does.

I also suggest 'directory' rather than 'folder' everywhere, as 'folder's can be non-directories like the network browser or control panel, but we really need them to be directories.

Reply:
I see you're already doing this in most places.

Reply:
> Given that we are describing to users what registries are, rather than describing what behavior something implementing registries must do, I think we should describe them with present tense, as in "registries contain a versions directory" rather than "registries must contain a versions directory". I think we should reserve 'must' for things the user has to do rather than things vcpkg does.

I agree with this recommendation, I'll revise the document.